sdk/services/kwentaToken.ts

```diff
@@ -0,0 +1,464 @@
+			KwentaToken,
+			KwentaStakingRewards,
+			vKwentaToken,
+			vKwentaRedeemer,
+			veKwentaToken,
+			veKwentaRedeemer,
+		} = this.sdk.context.contracts;
+
+		const map = {
+			kwenta: { contract: KwentaToken, spender: KwentaStakingRewards },
+			vKwenta: { contract: vKwentaToken, spender: vKwentaRedeemer },
+			veKwenta: { contract: veKwentaToken, spender: veKwentaRedeemer },
+		};
+
+		const { contract, spender } = map[token];
+
+		if (!contract || !spender) {
+			throw new Error(sdkErrors.UNSUPPORTED_NETWORK);
+		}
+
+		return this.sdk.transactions.createContractTxn(contract, 'approve', [spender.address, amount]);
+	}
+
+	public approveToken(
+		token: ContractName,
+		spender?: ContractName,
+		amount = ethers.constants.MaxUint256
+	) {
+		const tokenContract = this.sdk.context.contracts[token];
+
+		if (!tokenContract) {
+			throw new Error(sdkErrors.UNSUPPORTED_NETWORK);
+		}
+
+		let spenderAddress = this.sdk.context.walletAddress;
+
+		if (spender) {
+			const spenderContract = this.sdk.context.contracts[spender];
+			if (spenderContract) spenderAddress = spenderContract.address;
+		}
+
+		return this.sdk.transactions.createContractTxn(tokenContract, 'approve', [
+			spenderAddress,
+			amount,
+		]);
+	}
+
+	public redeemToken(token: ContractName) {
+		const tokenContract = this.sdk.context.contracts[token];
+
+		if (!tokenContract) {
+			throw new Error(sdkErrors.UNSUPPORTED_NETWORK);
+		}
+
+		return this.sdk.transactions.createContractTxn(tokenContract, 'redeem', [
+			this.sdk.context.walletAddress,
+		]);
+	}
+
+	public redeemVKwenta() {
+		return this.redeemToken('vKwentaRedeemer');
+	}
+
+	public redeemVeKwenta() {
+		return this.redeemToken('veKwentaRedeemer');
+	}
+
+	public vestToken(ids: number[]) {
+		const { RewardEscrow } = this.sdk.context.contracts;
+
+		if (!RewardEscrow) {
+			throw new Error(sdkErrors.UNSUPPORTED_NETWORK);
+		}
+
+		return this.sdk.transactions.createContractTxn(RewardEscrow, 'vest', [ids]);
+	}
+
+	public stakeKwenta(amount: string | ethers.BigNumber) {
+		return this.performStakeAction('stake', amount);
+	}
+
+	public async unstakeKwenta(amount: string | ethers.BigNumber) {
+		return this.performStakeAction('unstake', amount);
+	}
+
+	public async stakeEscrowedKwenta(amount: string | ethers.BigNumber) {
+		return this.performStakeAction('stake', amount, { escrow: true });
+	}
+
+	public async unstakeEscrowedKwenta(amount: string | ethers.BigNumber) {
+		return this.performStakeAction('unstake', amount, { escrow: true });
+	}
+
+	public async getClaimableRewards(periods: number[]) {
+		const { MultipleMerkleDistributor } = this.sdk.context.contracts;
+
+		if (!MultipleMerkleDistributor) {
+			throw new Error(sdkErrors.UNSUPPORTED_NETWORK);
+		}
+
+		const MultipleMerkleDistributorContract = new EthCallContract(
+			MultipleMerkleDistributor.address,
+			MultipleMerkleDistributorABI
+		);
+
+		const fileNames = periods
+			.slice(0, -1)
+			.map(
+				(i) =>
+					`trading-rewards-snapshots/${
```

Comment:
noticed this is done in two places, I think it would be nice to extract into a helper, would also clean things up a bit.

Reply:
Actually just realized we only use one of those two instances, so I deleted the other one (they were doing the exact same thing).